flows/runs/run.go

```diff
@@ -235,15 +276,16 @@ func NewRun(env flows.FlowEnvironment, flow flows.Flow, contact *flows.Contact,
 		flowUUID:    flow.UUID(),
 		contactUUID: contact.UUID(),
 		results:     flows.NewResults(),
-		environment: env,
 		contact:     contact,
 		flow:        flow,
 		status:      flows.StatusActive,
 		createdOn:   now,
 		modifiedOn:  now,
 	}
 
-	// create our new context
+	r.environment = newRunEnvironment(env, r)
```

Comment:
Are there any implications with run environments being stacked here? IE, subflows will be creating runs with an environment that is a runEnvironment.